fsck/fsck.c

```diff
@@ -780,17 +780,22 @@ static int read_file_dentry_set(struct exfat_de_iter *iter,
 	if (!node)
 		return -ENOMEM;
 
-	for (i = 2; i <= MIN(file_de->file_num_ext, 1 + MAX_NAME_DENTRIES); i++) {
+	name_de_count = DIV_ROUND_UP(stream_de->stream_name_len, ENTRY_NAME_MAX);
```

Comment:
It is better to calc `name_de_count` as:

```c
name_de_count = MAX(DIV_ROUND_UP(stream_de->stream_name_len, ENTRY_NAME_MAX), 1);
```

Reply:
Applied your comment and had pushed again.